src/NUnitEngine/nunit.engine/Services/DefaultTestRunnerFactory.cs

```diff
@@ -35,6 +35,16 @@ namespace NUnit.Engine.Services
     /// </summary>
     public class DefaultTestRunnerFactory : InProcessTestRunnerFactory, ITestRunnerFactory
     {
+        private IProjectService _projectService;
+
+        public override void StartService()
+        {
+            _projectService = ServiceContext.GetService<IProjectService>();
+            Status = _projectService != null && ((IService)_projectService).Status == ServiceStatus.Started
```

Comment:
If we know that `IProjectService` is an `IService`, shouldn't `IProjectService` derive from `IService` to that we don't have to do casting like this? If we don't know that we can cast safely here, then we should be checking if `_projectService is IService`. Either way, this cast feels wrong.